src/components/forms/SubmitTemplate/server-action.ts

```diff
@@ -0,0 +1,51 @@
+'use server';
+
+import { getSession, saveTemplate } from '@utils/form-actions';
+import { redirect, RedirectType } from 'next/navigation';
+import { templateFromSessionMapper, validateTemplate } from '@domain/templates';
+import { logger } from '@utils/logger';
+
+export async function submitTemplate(formData: FormData) {
+  const sessionId = String(formData.get('sessionId'));
+
+  const session = await getSession(sessionId);
+
+  if (!session) {
+    return redirect('/invalid-session', RedirectType.replace);
+  }
+
+  try {
+    if (session.templateType === 'UNKNOWN') {
+      throw new Error('Unknown template type');
+    }
+
+    const templateDTO = templateFromSessionMapper(
+      session.templateType,
+      session
+    );
+
+    const validatedTemplate = validateTemplate(
+      session.templateType,
+      templateDTO
+    );
+
+    const templateEntity = await saveTemplate(validatedTemplate);
+
+    // TODO: send email
+
+    // TODO: delete session
+
+    return redirect(
+      `/nhs-app-template-submitted/${templateEntity.id}`,
+      RedirectType.replace
+    );
+  } catch (error) {
+    logger.error('Failed to submit template', {
+      error,
+      sessionId,
+    });
+
+    // TODO: We need an internal server error page
+    throw error;
```

Comment:
Maybe remove this? 

I've intentionally left out logging on the above function calls and moved it up a level to be more of a catch-all.